### PR TITLE
Cancel builds was not cancelling all builds

### PR DESCRIPTION
**Pull request by ricardoamador**
When builds were being cancelled we would still find builds that were pending for a particular pull request. The logic here has been adjusted so that we simply look for builds that have been scheduled or are running and them cancel them. 

The previous logic would look through the builds and if there was a build that was not started, pending or scheduled it would return and not cancel anything. I am not sure why that was the logic though.

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes https://github.com/flutter/flutter/issues/124238

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/test/service/luci_build_service_test.dart [hunk 1]
```diff
@@ -584,15 +584,18 @@ void main() {
...
+
     test('Cancel builds when build list is empty', () async {
       when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
         return const BatchResponse(
           responses: <Response>[],
         );
       });
       await service.cancelBuilds(pullRequest, 'new builds');
+      // why is this getting called twice in this test?
       verify(mockBuildBucketClient.batch(any)).called(1);
```
Comment (ricardoamador) [inline app_dart/test/service/luci_build_service_test.dart:596]: I am still working on figuring out why this is getting called twice and not once. For some reason the batch function is getting called two times and not once.

File: app_dart/lib/src/service/luci_build_service.dart [hunk 2]
```diff
@@ -248,20 +248,25 @@ class LuciBuildService {
   ///
   /// Builds are queried based on the [RepositorySlug] and pull request number.
   Future<void> cancelBuilds(github.PullRequest pullRequest, String reason) async {
+    log.info(
+        'Attempting to cancel builds... for pullrequest ${pullRequest.base!.repo!.fullName}/${pullRequest.number}');
     final Map<String?, Build?> builds = await tryBuildsForPullRequest(pullRequest);
```
Comment (godofredoc) [inline app_dart/lib/src/service/luci_build_service.dart:253]: is tryBuildsForPullRequest reporting builds with pending status?

File: app_dart/lib/src/service/luci_build_service.dart [hunk 3]
```diff
@@ -248,20 +248,25 @@ class LuciBuildService {
...
+
     final List<Request> requests = <Request>[];
     for (Build? build in builds.values) {
-      requests.add(
-        Request(
-          cancelBuild: CancelBuildRequest(id: build!.id, summaryMarkdown: reason),
-        ),
-      );
+      // Scheduled status includes scheduled and pending tasks.
+      if (build!.status == Status.scheduled || build.status == Status.started) {
```
Comment (godofredoc) [inline app_dart/lib/src/service/luci_build_service.dart:258]: Should we also add pending here?

Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:258]: The Scheduled status includes both.

Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:258]: From this comment in the source: https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/model/luci/buildbucket.dart;l=862
I can certainly add it if that is what you mean.

File: app_dart/lib/src/service/luci_build_service.dart (hunk 2 above)
Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:253]: It is not specifying a state when it queries for the builds.

Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:253]: Okay we cannot get the status pending returned:
image

File: app_dart/lib/src/service/luci_build_service.dart (hunk 3 above)
Comment (ricardoamador) [reply app_dart/lib/src/service/luci_build_service.dart:258]: See my previous comment, it is not in the status enum.

**Comment by ricardoamador** [comment]
Okay I have addressed the outstanding comments and figured out the test.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**